src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int FIGURE_COUNT = 6;
+    public static final int MAX_SIDE_LENGTH = 101;
+    private static final Random random = new Random();
+    private static final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        Figure figure;
+        int figureType = random.nextInt(FIGURE_COUNT);
+        Color color = colorSupplier.getRandomColor();
+        if (figureType == 1) {
+            double radius = random.nextInt(1, MAX_SIDE_LENGTH);
+            figure = new Circle(radius, color);
+        } else if (figureType == 2) {
+            double firstBase = random.nextInt(1, MAX_SIDE_LENGTH);
+            double secondBase = random.nextInt(1, MAX_SIDE_LENGTH);
+            double height = random.nextInt(1, MAX_SIDE_LENGTH);
+            figure = new IsoscelesTrapezoid(firstBase, secondBase, height, color);
```

Comment:
1) you call random.nextInt(1, MAX_SIDE_LENGTH) too many times
I'd add 
private int getRandomSide() method with return random.nextInt(1, MAX_SIDE_LENGTH) and reused it in code

2) return may be used instead of writing to variable

3) switch/case is more readable than 3 if/elses :)
```suggestion
            return new IsoscelesTrapezoid(getRandomSide(), getRandomSide(), getRandomSide(), color);
```